README.md

```diff
@@ -690,44 +683,64 @@ spec:
     name: example-runners
   scaleUpTriggers:
-  - githubEvent:
-      checkRun:
-        types: ["created"]
-        status: "queued"
-    amount: 1
-    duration: "5m"
+    - githubEvent:
+        workflowJob: {}
+      duration: "30m"
 ```
 
-With the above example, the webhook server scales `example-runners` by `1` replica for 5 minutes on each `check_run` event with the type of `created` and the status of `queued` received.
+The lifecycle of a runner provisioned from a webhook is different to a runner provisioned from the pull based scaling method:
 
-Of note is the `HRA.spec.scaleUpTriggers[].duration` attribute. This attribute is used to calculate if the replica number added via the trigger is expired or not. On each reconciliation loop, the controller sums up all the non-expiring replica numbers from previous scale-up triggers. It then compares the summed desired replica number against the current replica number. If the summed desired replica number > the current number then it means the replica count needs to scale up.
+1. GitHub sends a `workflow_job` event to ARC with `status=queued`
+2. ARC finds HRAs with a `workflow_job` webhook scale trigger that and back a RunnerDeployment / RunnerSet with matching runner labels
+3. The matched HRAs adds a unit to its `capacityReservations` list
+4. ARC adds a replica and sets the EffectiveTime of that replica to current + `HRA.spec.scaleUpTriggers[].duration`
 
-As mentioned previously, the `scaleDownDelaySecondsAfterScaleOut` property has the final say still. If the latest scale-up time + the anti-flapping duration is later than the current time, it doesn’t immediately scale down and instead retries the calculation again later to see if it needs to scale yet.
+At this point there are a few things that can happen, either the job gets allocated to the runner or the runner is left dangling due to it not being used, if the runner gets assigned the job that triggered the scale up the lifecycle looks like this:
 
----
+1. The new runner gets allocated the job and processes it
+2. Upon the job ending GitHub sends another `workflow_job` event to ARC but with `status=completed`
+3. The HRA removes a capacity reservation from its `capacityReservations` and picks a runner to terminate ensuring it isn't busy via the GitHub API beforehand
+
+If the job is cancelled before it is allocated to a runner or the runner is never used due to other runners matching needed runner group and required runner labels are allocated the job then the lifecycle looks like this:
 
-The primary benefit of autoscaling on Webhooks compared to the pull driven scaling is that it is far quicker as it allows you to immediately add runner resources rather than waiting for the next sync period.
+1. The new runner is retained until its `EffectiveTime` passes
+3. The HRA removes a capacity reservation from its `capacityReservations` and terminates the expired runner ensuring it isn't busy via the GitHub API beforehand
```

Comment:
```suggestion
1. The HRA removes a capacity reservation from its `capacityReservations` and terminates the expired runner ensuring it isn't busy via the GitHub API beforehand
```

`EffectiveTime` is used to prevent creating a redundant runner, in case a runner pod has successfully finished running a job and completed with exist status 0. So I'd say it isn't involved in this scenario.

`EffectiveTime` works as follows:

There is a race between the timing that the runner pod finishes running and the timing that GitHub sends a `status=completed` `workflow_job` event.

When the former happened before the latter(which is almost 100% cases), ARC used to end up with recreating a runner pod because RunnerDeployment specs > available runner pod replicas.

With EffectiveTime and saving the EffectiveTime that ARC saw when it ran the last reconciliation on the RunnerDeployment, ARC can prevent this pod recreation because it now knows that it already created all the desired replicas after seeing that EffectiveTime.